src/sync/installation.ts

```diff
@@ -49,9 +57,9 @@ const updateNumberOfReposSynced = async (
 		// all 3 statuses need to be complete for a repo to be fully synced
 		const { pullStatus, branchStatus, commitStatus } = repos[id];
 		return (
-			pullStatus === "complete" &&
-			branchStatus === "complete" &&
-			commitStatus === "complete"
+			completedTaskStatuses.includes(pullStatus)
+			&& completedTaskStatuses.includes(branchStatus)
+			&& completedTaskStatuses.includes(commitStatus)
```

Comment:
To mark a repo as "synced", we now have to consider the "complete" and "failed" statuses instead of just the "complete" status.